src/main/java/at/hannibal2/skyhanni/features/garden/composter/ComposterOverlay.kt

```diff
@@ -414,8 +418,35 @@ class ComposterOverlay {
             list.add(Renderable.link("$name§r §8x${itemsNeeded.addSeparators()} §7(§6$format§7)") {
                 onClick(internalName)
                 if (LorenzUtils.isControlKeyDown()) {
-                    inInventory = false
-                    BazaarApi.searchForBazaarItem(itemName, itemsNeeded.toInt())
+                    if(lastAttemptTime.passedSince()>0.5.seconds){
+                        lastAttemptTime = SimpleTimeMark.now()
+                        val amountInSacks = fetchSackItem(internalName.asInternalName())?.amount
+                        val sacksLoaded = fetchSackItem(internalName.asInternalName())?.outdatedStatus
+                        if (itemsNeeded.toInt() != 0 && itemName.removeColor() != "Biofuel") {
+                            if (config.composterOverlayGetType == 0) {
+                                inInventory = false
+                                BazaarApi.searchForBazaarItem(itemName, itemsNeeded.toInt())
+                            } else if (amountInSacks == 0 && sacksLoaded != -1) {
+                                SoundUtils.createSound("mob.endermen.portal", 0F).playSound()
+                                chat("§e[SkyHanni] No $itemName §r§efound in sacks. Opening Bazaar.")
+                                inInventory = false
+                                BazaarApi.searchForBazaarItem(itemName, itemsNeeded.toInt())
+                            } else {
+                                val having = InventoryUtils.countItemsInLowerInventory { it.getInternalName() == internalName.asInternalName() }
+                                if (sacksLoaded == -1) {
+                                    clickableChat("§e[SkyHanni] sacks could not be loaded. Open your sacks to update the data!.","sax")
+                                }
+                                if (amountInSacks != null && amountInSacks < itemsNeeded) {
+                                    clickableChat("§e[SkyHanni] You're out of $itemName §r§ein your sacks! Click here to buy on the Bazaar!","bz ${itemName.removeColor()}")
+                                }
+                                if (having < itemsNeeded) {
+                                    LorenzUtils.sendCommandToServer("gfs $internalName ${itemsNeeded.toInt() - having}")
+                                } else {
+                                    chat("§e[SkyHanni] $itemName §r§8x${itemsNeeded.toInt()}§r§e already found in inventory!")
```

Comment:
i dont see why you need to reset the formatting in this string "§r"

Reply:
A new color resets the formatting as well